umh-core/pkg/communicator/pkg/tools/watchdog/watchdog.go

```diff
@@ -295,14 +295,19 @@ func (s *Watchdog) ReportHeartbeatStatus(uniqueIdentifier uuid.UUID, status Hear
 	var warnings uint32
 	onlyIfHasSub := hb.onlyIfSubscribers
 	hasSubs := s.hasSubscribers.Load()
-	if status == HEARTBEAT_STATUS_WARNING {
+	switch status {
+	case HEARTBEAT_STATUS_WARNING:
 		warnings = hb.warningCount.Add(1)
 		if s.warningsAreErrors.Load() {
 			s.logger.Errorf("[%s] Heartbeat %s (%s) send a warning (%d/%d) and warnings are errors", s.watchdogID, name, uniqueIdentifier, warnings, hb.warningsUntilFailure)
 			s.badHeartbeatChan <- uniqueIdentifier
 		}
-	} else if status == HEARTBEAT_STATUS_OK {
+	case HEARTBEAT_STATUS_OK:
 		hb.warningCount.Store(0)
+	case HEARTBEAT_STATUS_ERROR:
```

Comment:
This changes the semantic!
Previously the warnings check was done before the error check.

Reply:
what is the targeted order of executing this? because I dont see why it should matter at all

Reply:
see slack discussion